### PR TITLE
Refactor MCP kernel discovery and add unit tests for core helpers

### PR DESCRIPTION
**Pull request by irvineoy**
## Summary

This PR improves both code organization and test coverage across Magpie.

It extracts MCP kernel discovery logic out of `Magpie/mcp/server.py` into a dedicated `Magpie/mcp/discovery.py` module, tightens Triton kernel detection to avoid treating arbitrary `.py` files as Triton kernels, and adds a first broader set of unit tests for core non-GPU helper logic across the project.

## What Changed

- Refactored MCP kernel discovery into `Magpie/mcp/discovery.py`
- Kept `Magpie/mcp/server.py` focused on MCP tool wiring and response handling
- Improved Triton discovery so only actual Triton-decorated Python kernel files are returned
- Skipped common generated/output directories during discovery such as `build`, `dist`, `bin`, `out`, and `results`
- Added unit tests for:
  - kernel/config parsing in `Magpie.main`
  - `KernelEvalConfig` command normalization and deserialization
  - benchmark config/dataclass behavior
  - legacy TraceLens config parsing
  - image selection logic
  - benchmark result parsing
  - common environment/compile helpers
  - remote task helper logic
- Fixed a bug surfaced by the new tests:
  - `TraceLensConfig.from_dict()` now correctly handles legacy configs that only provide `export_excel`

## Why

Before this PR:
- `discover_kernels(..., kernel_type="triton")` could overmatch unrelated Python files
- MCP server logic and discovery logic were more tightly coupled than necessary
- the repo had very limited unit test coverage for core helper and parsing logic
- legacy TraceLens config handling had an edge case that silently produced the wrong export format

This PR makes discovery more accurate, the MCP layer easier to maintain, and the project safer to change going forward.

## Testing

Ran:

```bash
.venv/bin/python -m pytest -q

### CODE REVIEW

File: Magpie/mcp/discovery.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        "__pycache__",
+        ".cache",
+        "venv",
+        ".venv",
+        "env",
+        ".env",
+        ".tox",
+        ".nox",
+        ".pytest_cache",
+        "build",
```
Comment (sinarafati-amd) [inline Magpie/mcp/discovery.py]: if I am not wrong CMake-generated Triton files might show up in build folder so probably we should not include this as skipping dir

Comment (irvineoy) [reply Magpie/mcp/discovery.py]: Thanks, good catch. I updated discovery so build-like directories are no longer skipped for Triton, which allows generated Triton kernels to be found, while still filtering out non-Triton/generated noise there. I also added tests for both behaviors.

**Review by haofrank — APPROVED** [review]
LGTM
